src/lib/alloc.c

```diff
@@ -697,12 +712,70 @@ void rfree(void *ptr)
 	}
 
 	/* free the block */
-	spin_lock_irq(&memmap.lock, flags);
 	free_block(ptr);
-	spin_unlock_irq(&memmap.lock, flags);
 	memmap.heap_trace_updated = 1;
 }
 
+void rfree(void *ptr)
+{
+	uint32_t flags;
+
+	/* sanity check - NULL ptrs are fine */
+	if (!ptr)
+		return;
+
+	/* free the block */
+	spin_lock_irq(&memmap.lock, flags);
+	_rfree_unlocked(ptr);
+	spin_unlock_irq(&memmap.lock, flags);
+}
+
+void *_realloc(void *ptr, int zone, uint32_t caps, size_t bytes)
```

Comment:
I would add an arg here to specify if we need to copy over old data. I would imagine buffers wont need copy of old data.

Reply:
ok